### PR TITLE
Update galaxy_ng container for tests

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Update galaxy_ng container for tests

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
many

##### ADDITIONAL INFORMATION
Blocked by:

- ~https://github.com/pulp/pulp_ansible/pull/1432~
- ~https://issues.redhat.com/browse/AAH-2329~
- ~https://issues.redhat.com/browse/AAH-2332~
- https://issues.redhat.com/browse/AAH-2328

AAH-2328 is still open, but there is a workaround in that right now the admin user can bypass content approval, so it's enabled to work around the errant behavior, but should really be disabled once that bug eventually gets fixed.

### CODE REVIEW

File: test/integration/targets/ansible-galaxy-collection/library/setup_collections.py
Comment (jborean93) [inline test/integration/targets/ansible-galaxy-collection/library/setup_collections.py:268]: Is keeping the `start`/`end`/`delta` still desired? I know it doesn't really matter but curious if the newer version is slower or temperamental still?

File: test/integration/targets/ansible-galaxy-collection/tasks/publish.yml
Comment (jborean93) [inline test/integration/targets/ansible-galaxy-collection/tasks/publish.yml:9]: Minor nit, this is a pretty long line, would it be possible to do `>-`?

File: test/integration/targets/ansible-galaxy-collection/tasks/supported_resolvelib.yml
Comment (jborean93) [inline test/integration/targets/ansible-galaxy-collection/tasks/supported_resolvelib.yml:26]: Any reason why we have both `test_server` and `test_api_server`?

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env
Comment (jborean93) [inline test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env]: Wouldn't AH actual be auto signing collections?

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env
Comment (jborean93) [inline test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env]: Does this not break the publish tests as now they won't be downloadable until approved?

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py
Comment (jborean93) [inline test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:84]: Do the tests now pass with https://github.com/ansible/ansible/blob/695d1065034c282becedf94dd4f122babce61964/test/integration/targets/ansible-galaxy-collection/tasks/install.yml#L703 at this newer version?

File: test/integration/targets/ansible-galaxy-collection/library/setup_collections.py
Comment (sivel) [reply test/integration/targets/ansible-galaxy-collection/library/setup_collections.py:268]: I added it for purposes of adjusting the timeout defined higher in this module. I figured leaving this in this could help address problems later.

File: test/integration/targets/ansible-galaxy-collection/tasks/supported_resolvelib.yml
Comment (sivel) [reply test/integration/targets/ansible-galaxy-collection/tasks/supported_resolvelib.yml:26]: It's not really useful here, and I could have just done `test_api_server|default(test_server)` in `install.yml` but I just stuck with being explicit.  The `test_server` isn't functional to make an API request to get details about the collection.  So the v2 tests, need to make some `uri` calls to a `v3` API due to limitations of the v2 API in `pulp_ansible`.

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env
Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env]: Yes, and that should be a future enhancement to these tests.  Ultimately we will need to reconcile how to get signing functional.  I was given this URL for when we plan to attempt this: https://github.com/pulp/oci_env/blob/main/profiles/galaxy_base/init.sh

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env
Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.env]: In short yes, but it doesn't break the publish tests for our current use case.  Ideally we should have this disabled, but due to the linked bug that is not possible.  In any case, this setting when set to `true` does not apply to the `admin` user.  So anything the admin user publishes doesn't require approval.

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py
Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:84]: The image used in this PR is the newest available, with these versions in it:

```
galaxy-importer==0.4.10
galaxy-ng==4.7.1
pulp-ansible==0.17.2
pulp-container==2.14.5
pulp-glue==0.19.2
pulpcore==3.23.6
```

Comment (jborean93) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:84]: I just recall I had to comment out the explicit tests in the file mentioned because of a bug in the latest version snapshot with caching at the time. Now that it's updated do the commented out tests now pass?

Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:84]: Ah I see, let me check that

Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:84]: Ok, uncommented tests, but restricted them to just `galaxy_ng`.

**Review by jborean93 — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/cli/galaxy.py
Comment (s-hertel) [inline lib/ansible/cli/galaxy.py:82]: I don't think this was ever documented, but since there was no internal-only warning, is this worth mentioning in a changelog?

Comment (sivel) [reply lib/ansible/cli/galaxy.py:82]: Done

File: setup.cfg
Comment (mattclay) [inline setup.cfg]: Could this be included inline in the galaxy test plugin (perhaps as a dictionary), and written to a temp file when needed? That would avoid having to list the file here, and would also work with change detection without additional modifications.

If not, we should update change detection to recognize this file as part of the associated test plugin. I'd also consider moving it under `test/lib/ansible_test/_util/controller/`, to avoid having non-Python files in the `test/lib/ansible_test/_internal/` directory tree.

File: test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py
Comment (mattclay) [inline test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:146]: Are these files not needed before the container is started, to avoid a race condition?

Comment (sivel) [reply test/lib/ansible_test/_internal/commands/integration/cloud/galaxy.py:146]: No, they are not needed before container start. The previous configuration needed to exist before start up, but since this one file is only read when `galaxy-importer` is called, and not by galaxy/pulp directly, it can be injected afterward.

File: setup.cfg
Comment (sivel) [reply setup.cfg]: It could be.  During testing it was easiest for me to develop it as a standalone file, so I could use both via `ansible-test` and when running the container manually via `docker run`.

Comment (sivel) [reply setup.cfg]: Done.

**Review by mattclay — APPROVED** [review]

**Pull request merged**
